addons/gloot/ctrl_inventory_grid.gd

```diff
@@ -314,18 +334,62 @@ func _select(item: InventoryItem) -> void:
     _selected_item = item
     _refresh_selection()
 
+func switch_virtual_cursor_grid(new_grid):
+    if not _grabbed_with_virtual_cursor or not new_grid:
+        return
+
+    _virtual_cursor_gird = new_grid
+    _virtual_cursor = new_grid.rect_global_position
+
 
 func _input(event: InputEvent) -> void:
-    if !(event is InputEventMouseButton):
+    if event.is_action_released("ui_cancel"):
+        _cancel_grab()
         return
 
-    if !_grabbed_ctrl_inventory_item:
+    _handle_virtual_cursor(event)
+    _handle_mouse(event)
+
+func _handle_virtual_cursor(event: InputEvent):
+    if not _grabbed_with_virtual_cursor:
+        return
+
+    var delta
+    if event.is_action_pressed("ui_left"):
+        delta = Vector2(-_virtual_cursor_gird.field_dimensions.x, 0)
```

Comment:
I'm afraid we need to take `item_spacing` into account here (same for the other cases below).